pkgs/build-support/release/maven-build.nix

```
@@ -12,7 +12,7 @@
 } @ args :
 
 let
-  mvnFlags = "-Dmaven.repo.local=$M2_REPO ${if doTest then "" else "-Dmaven.test.skip.exec=true"} ${extraMvnFlags}";
+  mvnFlags = builtins.toString [ "-Dmaven.repo.local=$M2_REPO" "${if doTest then "" else "-Dmaven.test.skip.exec=true"}" "${extraMvnFlags}" ];
```

Maybe something like:

```suggestion
  mvnFlags = builtins.toString ([
    "-Dmaven.repo.local=$M2_REPO"
    (lib.optionalString (!doTest) "-Dmaven.test.skip.exec=true")
  ] ++ extraMvnFlags);
```

---

`extraMvnFlags` is a string but github search shows only one non-nixpkgs usage of it so it should be fine to change it to a list 
https://github.com/search?p=6&q=extraMvnFlags&type=Code
https://github.com/taktoa/nix-packages/blob/master/packages/kframework/default.nix
@taktoa